text/0732-cdk-cli-telemetry.md

```diff
@@ -0,0 +1,140 @@
+      "parameters": {},
+      "config": {},
+    },
+  }
+}
+```
+
+Alternatively, if deployment fails, the deploy **event** looks like this:
+
+```json
+{
+  "identifiers": {
+    "sessionId": "14B36D48-4DFF-47C3-B0E4-D966DD6DB038",
+    "eventId": "14B36D48-4DFF-47C3-B0E4-D966DD6DB038:1",
+    "installationId": "3F1FD23A-58A9-4C0D-8A82-098D6101B322",
+    "accountId": "DDCA70E4-C73A-4B93-B464-BBA7DCCB6B86",
+    "region": "us-east-1",
+  },
+  "event": {
+    "state": "FAIL",
+    "eventType": "deploy",
+    "command": {
+      "path": ["cdk", "deploy", "<REDACTED>"],
+      "parameters": {},
+      "config": {},
+    },
+  },
+  "error": {
+    "name": "ToolkitError",
+    "message": "Error: Asset <REDACTED> upload failed: AccessDenied: Access Denied",
+    "trace": "    at AssetPublishing.publishAsset (<PATH>/aws-cdk/lib/assets/asset-publishing.js:128:23)
+    at CloudFormationDeployment.publishAssets (<PATH>aws-cdk/lib/api/cloudformation-deployment.js:295:41)
+    at CloudFormationStackArtifact.prepareForDeployment (<PATH>aws-cdk/lib/api/cdk-toolkit.js:517:12)"
+    "logs": "Deploying stack <REDACTED>
+    IAM Statement Changes
+    ┌───┬─────────────────────────┬────────┬─────────────────────────┬─────────────────────────┬───────────┐
+    │   │ Resource                │ Effect │ Action                  │ Principal               │ Condition │
+    ├───┼─────────────────────────┼────────┼─────────────────────────┼─────────────────────────┼───────────┤
+    │ + │ <REDACTED>              │ Allow  │ sts:AssumeRole          │ Service:lambda.amazonaw │           │
+    │   │                         │        │                         │ s.com                   │           │
+    └───┴─────────────────────────┴────────┴─────────────────────────┴─────────────────────────┴───────────┘
+    (NOTE: There may be security-related changes not in this list. See https://github.com/aws/aws-cdk/issues/1299)
+
+    Bundling asset <PATH>...",
+  }
+}
+```
+
+## Opt-Out
+
+Collecting anonymous data will help the CDK team on two fronts: usage metrics and error debugging.
+Both of these aspects will help us better serve CDK customers in the future.
+We will ensure that we are not collecting [customer content](https://aws.amazon.com/compliance/data-privacy-faq/#topic-1)
+or personally-identifiable information.
+
+You can easily opt-out of sending telemetry data.
+Customers can set `toolkit-telemetry: false` in their `cdk.json` configuration files.
+Setting `toolkit-telemetry: false` in the CDK App level `cdk.json` will disable telemetry in your CDK App.
+Setting `toolkit-telemetry: false` in the `~/.cdk.json` file will disable telemetry across all CDK Apps on the same machine.
+
+We will also respect an environment variable, `TOOLKIT_TELEMETRY=false`. If set, this will achieve the same effect of disabling telemetry.
+
+Alternatively, you can run the CLI command (which disables telemetry for the CDK App):
+
+```bash
+> cdk telemetry --disable
+```
```

Comment:
Looking into the future, I'd be nice to write this to be able to write this as well:

```
cdk deploy --no-toolkit-telemetry`
```

Reply:
as a global variable right? i agree. do you think it's necessary to have this immediately or are the other opt-out methods sufficient for now